redis/src/main/scala/zio/redis/Interpreter.scala

```diff
@@ -60,35 +62,39 @@ trait Interpreter {
         Promise.make[RedisError, String].flatMap(p => pendingRequests.offer(Request(command, p)).as(p))
 
       private val executeNext: UIO[Any] =
-        pendingRequests.take.flatMap { request =>
+        pendingRequests.take.map { request =>
           val exchange =
-            IO {
-              unsafeSend(request.command)
-              unsafeReceive()
+            IO.effect {
+              try {
+                unsafeSend(request.command)
+                unsafeReceive()
+              } catch {
+                case t: Throwable => throw RedisError.ProtocolError(t.getMessage)
+              } finally readinessFlag.set(true)
             }
 
-          val completer = exchange.catchAll(RedisError.make).ensuring(UIO(readinessFlag.set(true)))
-
-          request.result.completeWith(completer)
+          request.result.unsafeDone(exchange.refineToOrDie[RedisError])
```

Comment:
Technically we are doing an effect inside `map`, which is a no-no, but I think this one is safe to do and gives us a performance boost, so 👍